### ISSUE TITLE
Replace raw pointers by smart pointers

### ISSUE THREAD

**Issue by bit-shift** [kr15h/ofxPiMapper#82]
Raw pointers are error-prone and should be replaced by std::*_ptr, depending on the use case. It's also easier to reason about the data-flow of the application if shared / unique and weak pointers are used.

**Comment by kr15h** [comment]
Yes, it's on my mental to-do list already : ] Thanks for opening an issue.

**Comment by bit-shift** [comment]
I will work on this. More issues to come :)

**Comment by kr15h** [comment]
Ok : ] Cool. Very cool!

**Comment by bit-shift** [comment]
Currently, I don't have the time to work on this any further.
